.github/workflows/ci.yml

```diff
@@ -17,6 +17,8 @@ jobs:
         include:
           - name: default
             command: make -f misc/docker-ci.mk
+          - name: asan
+            command: make -f misc/docker-ci.mk CMAKE_ENVS='CC=clang CXX=clang++ CFLAGS="-fsanitize=address,undefined" CXXFLAGS="-fsanitize=address,undefined" LDFLAGS="-fsanitize=address,undefined"' CHECK_ENVS="ASAN_OPTIONS=detect_leaks=0"
```

Comment:
As said on https://github.com/h2o/h2o/pull/2891, I would like to agree on the intent: is it to turn on both Asan and UBsan and fix all the issues? Do we need to both at once?

Reply:
No, it would be nicer for us to focus on ASan first. 

When we add UBSan, we should add `UBSAN_OPTIONS=print_stacktrace=1:halt_on_error=1` as a part of `CHECK_ENVS` to make sure no UB happens. Otherwise adding `-fsanitize=undefined` doesn't make sense.